includes/class-sensei-course.php

```diff
@@ -4336,6 +4387,9 @@ public function remove_legacy_course_actions() {
 		// Course prerequisite completion message.
 		remove_action( 'sensei_single_course_content_inside_before', [ 'Sensei_Course', 'prerequisite_complete_message' ], 20 );
 
+		// Self enrollment not allowed message.
+		remove_action( 'sensei_single_course_content_inside_before', [ 'Sensei_Course', 'self_enrollment_not_allowed_message' ], 20 );
```

Comment:
Similar to the other "legacy" hooks (for courses not using blocks), it removes the message, since it's added through the Take Course block.